compiler/src/dotty/tools/dotc/transform/Splicing.scala

```diff
@@ -242,9 +242,14 @@ class Splicing extends MacroTransform:
                 else args.mapConserve(arg => transformLevel0QuoteContent(arg)(using quoteContext))
               }
               cpy.Apply(tree)(cpy.Select(sel)(cpy.Apply(app)(fn, newArgs), nme.apply), quotesArgs)
-        case Apply(TypeApply(_, List(tpt)), List(quotes))
+        case Apply(TypeApply(typeof, List(tpt)), List(quotes))
         if tree.symbol == defn.QuotedTypeModule_of && containsCapturedType(tpt.tpe) =>
-          ref(capturedType(tpt))(using ctx.withSource(tree.source)).withSpan(tree.span)
+          tpt match
+            case block: Block =>
+              val newBlock = capturedBlockPartTypes(block)
+              Apply(TypeApply(typeof, List(newBlock)), List(quotes)).withSpan(tree.span)
+            case _ =>
+              ref(capturedType(tpt))(using ctx.withSource(tree.source)).withSpan(tree.span)
```

Comment:
In the following case 
```scala
import scala.quoted.*

trait Api:
  type Reader

def bugImpl[T: Type](using Quotes) =
  '{
    val p: Api = ???
    ${
      Type.of[List[p.Reader]]
      Expr(1)
    }
  }
```

we get a hole with `{{{ 0 | Int | List[p.Reader] | ...`. I would have expected to have `p.Reader` instead of `List[p.Reader]` as your code does for the other examples in the test. 

It seems that if we would apply the same `TypeMap` we have in `capturedBlockPartTypes` we could make this happen. We can probably addapt `capturedBlockPartTypes` to handle both cases.